docs/DeveloperGuide.md

```diff
@@ -1,38 +1,275 @@
 
-## Design & implementation
+###### [Back to table of contents](#table-of-contents)
+
+## Setting up, getting started
+
+1. Java 11 is required for Meal Companion to function properly, please make sure your device has Java 11 installed. If not, you may download it [here](https://www.oracle.com/sg/java/technologies/downloads/#java11). If you need further asssistance installing Java 11, you may refer to the [Installation Guide](https://docs.oracle.com/en/java/javase/11/install/installation-jdk-microsoft-windows-platforms.html).
+
+2. Once you have ascertained that your device has Java 11 installed, you may download the `.jar file` [here](https://github.com/AY2223S2-CS2113T-T09-3/tp/releases/tag/v2.0). 
+
+![jar2.png](images/jar2.png)
+
+Click on `mealcompanion.jar` to automatically download the file. Place the file in a folder of your choice.
+
+If you wish to view the source code, you may click on either `Source code (zip)` or `Source code (tar.gz)` to download the source code of Meal Companion for your perusal and testing. Unzip the folder and open the project using any Integrated Development Environment (IDE), although we recommend using IntelliJ Idea. You can download IntelliJ Idea [here](https://www.jetbrains.com/idea/download/#section=windows). Please ensure you download the right version for your respective operating system. 
+
+3. Open the command terminal and navigate to the folder where you downloaded `mealcompanion.jar` using `cd <file path>` commands as necessary.
+4. Ensure the terminal is within the correct folder.
+5. Use the command `java -jar mealcompanion.jar` to launch Meal Companion.
+![startup1.png](images/startup1.png)
+6. When you are finished using Meal Companion, use the `bye` command to terminate the application. This will ensure your stored data is saved correctly for future access.
+
+Please be advised that we cannot guarantee complete functionality of Meal Companion if you have modified the source code in your own local IDE. If you have any queries, feel free to reach out to any member of the development team and we will do our best to assist you.
+
+###### [Back to table of contents](#table-of-contents)
+
+## Design
+
+### Command Parsing
+
+The system of `Routable` classes is built to allow for a composable setup of commands and subcommands.
+
+Developers are able to build a "routing tree" by nesting `CommandRouterNode`s together, allowing for command matching
+to be done in a robust and extensible way. This architecture was chosen because our commands are designed to have common
+prefixes, such as the "recipe all" and "recipe possible" commands.
+
+Therefore, we need to standardize the way we parse
+the command in such a way that we do not have to worry about match priority or ordering, as we would if we
+were to implement this with simple if-else or switch statements.
+
+![RouterUML.png](images/RouterUML.png)
```

Comment:
For UML methods, the return type should be at the back of the method instead of at the back. For example, `getRouteName(): String`. Maybe you can check it out?